openedx/core/djangoapps/embargo/tests/test_views.py

```diff
@@ -85,3 +94,52 @@ def _load_page(self, access_point, message_key, expected_status=200):
                 actual=response.status_code
             )
         )
+
+
+@skip_unless_lms
+class CheckCourseAccessViewTest(CourseApiFactoryMixin, ModuleStoreTestCase):
+    """ Tests the course access check endpoint. """
+
+    @patch.dict(settings.FEATURES, {'EMBARGO': True})
+    def setUp(self):
+        super(CheckCourseAccessViewTest, self).setUp()
+        self.url = reverse('api_embargo:v1_course_access')
+        user = self.create_user("staff", is_staff=True)
+        self.client.login(username=user.username, password='edx')
+        self.course_id = str(CourseFactory().id)
+        self.request_data = {
+            'course_ids': [self.course_id],
+            'ip_address': '0.0.0.0',
+            'user': self.user,
+        }
+
+    def test_course_access_endpoint_with_unrestricted_course(self):
+        response = self.client.get(self.url, data=self.request_data)
+        expected_response = {'Access': True}
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data, expected_response)
+
+    def test_course_access_endpoint_with_restricted_course(self):
+        CountryAccessRuleFactory(restricted_course=RestrictedCourseFactory(course_key=self.course_id))
+
+        self.user.is_staff = False
+        self.user.save()
+        # Appear to make a request from an IP in the blocked country
+        with mock.patch.object(pygeoip.GeoIP, 'country_code_by_addr') as mock_ip:
+            mock_ip.return_value = 'US'
+            response = self.client.get(self.url, data=self.request_data)
+        expected_response = {'Access': False}
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data, expected_response)
+
+    def test_course_access_with_logged_out_user(self):
+        self.client.logout()
+        response = self.client.get(self.url, data=self.request_data)
+        self.assertEqual(response.status_code, 403)
+
+    def test_course_access_with_non_staff_user(self):
+        user = self.create_user('non-staff', is_staff=False)
+        self.client.login(username=user.username, password='edx')
```

Comment:
Don't hardcode the password. I'm pretty certain this is stored in a constant someplace. Use it.